TestFoundation/TestNSGeometry.swift

```diff
@@ -447,4 +452,157 @@ class TestNSGeometry : XCTestCase {
+        
+        // Random spaces
+        stringPoint = "{  0.6   , 5.4   }"
+        stringSize = "{ 105.7, 24.3       }"
+        stringRect = "{{0.6 , 5.4}   ,{105.7 ,24.3}}"
+        
+        point = NSPointFromString(stringPoint)
+        XCTAssertEqual(expectedPoint, point,
+                       "\(NSStringFromPoint(point)) is not equal to expected \(NSStringFromPoint(expectedPoint))")
+        
+        size = NSSizeFromString(stringSize)
+        XCTAssertEqual(expectedSize, size,
+                       "\(NSStringFromSize(size)) is not equal to expected \(NSStringFromSize(expectedSize))")
+        
+        rect = NSRectFromString(stringRect)
+        XCTAssertEqual(expectedRect, rect,
+                       "\(NSStringFromRect(rect)) is not equal to expected \(NSStringFromRect(expectedRect))")
+    
+    }
+    
+    func test_DecodeEmptyStrings() {
+        let stringPoint = ""
+        let stringSize = ""
+        let stringRect = ""
+        
+        let expectedPoint = NSZeroPoint
+        let point = NSPointFromString(stringPoint)
+        XCTAssertEqual(expectedPoint, point,
+                       "\(NSStringFromPoint(point)) is not equal to expected \(NSStringFromPoint(expectedPoint))")
+
+        let expectedSize = NSZeroSize
+        let size = NSSizeFromString(stringSize)
+        XCTAssertEqual(expectedSize, size,
+                       "\(NSStringFromSize(size)) is not equal to expected \(NSStringFromSize(expectedSize))")
+        
+        let expectedRect = NSZeroRect
+        let rect = NSRectFromString(stringRect)
+        XCTAssertEqual(expectedRect, rect,
+                       "\(NSStringFromRect(rect)) is not equal to expected \(NSStringFromRect(expectedRect))")
+    }
+    
+    func test_DecodeNegativeFromNSString() {
+        let stringPoint = "{-0.6, -5.4}"
+        let stringSize = "{-105.7, -24.3}"
+        let stringRect = "{{-0.6, -5.4}, {-105.7, -24.3}}"
+        
+        let expectedPoint = NSMakePoint(CGFloat(-0.6), CGFloat(-5.4))
+        let point = NSPointFromString(stringPoint)
+        XCTAssertEqual(expectedPoint, point,
+            "\(NSStringFromPoint(point)) is not equal to expected \(NSStringFromPoint(expectedPoint))")
+        
+        let expectedSize = NSMakeSize(CGFloat(-105.7), CGFloat(-24.3))
+        let size = NSSizeFromString(stringSize)
+        XCTAssertEqual(expectedSize, size,
+                       "\(NSStringFromSize(size)) is not equal to expected \(NSStringFromSize(expectedSize))")
+        
+        let expectedRect = NSMakeRect(CGFloat(-0.6), CGFloat(-5.4), CGFloat(-105.7), CGFloat(-24.3))
+        let rect = NSRectFromString(stringRect)
+        XCTAssertEqual(expectedRect, rect,
+                       "\(NSStringFromRect(rect)) is not equal to expected \(NSStringFromRect(expectedRect))")
+        
+    }
+    
+    func test_DecodeGarbageFromNSString() {
+        var stringRect = "-0.6a5.4das-105.7bfh24.3dfas;hk312}}"
```

Comment:
Garbage is now considered as a separator. Couldn't make it work with exponential format, though. Not sure if it should be supported on `NSGeomerty` side, probably it should be implemented in `NSScanner`.